js/ui/text_box/text_box.js

```diff
@@ -94,6 +94,20 @@ const TextBox = TextEditor.inherit({
         this._$searchIcon = $searchIcon;
     },
 
+
+    _renderLabel: function() {
```

Comment:
So, why not to extract an animated label as a separate component? 
We will be able to collect the logic of the label in one place, and specific widgets, if necessary, will switch its individual options

Reply:
What is the difference between placeholder and label? Placeholder is not the separate component. But I agree to use renderLabel method with arguments for lookup and tagbox widgets.